Add service/instance/managed labels to HPAs. PAASTA-18226

EvanKrall opened the pull request:
This should make the kube_horizontalpodautoscaler_labels metric from kube-state-metrics more useful.

nemacysts commented on paasta_tools/kubernetes_tools.py:
```diff
@@ -910,10 +910,17 @@ def get_autoscaling_metric_spec(
             autoscaling_params,
         )
+
+        labels = {
```
might be nice to add the pool here too - then we can easily grab all the HPAs for a given pool

EvanKrall commented:
I added pool and refactored the tests to use mock.ANY for labels instead of hard-coding the labels, since these tests aren't meant to test the labels.
